tutorials/text/llama-nemotron-data-curation/utils/jsonl_utils.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+import os
+from collections.abc import Generator
+
+import ray
+
+
+@ray.remote
+def split_jsonl_by_size(input_file: str, target_size_mb: int, output_dir: str, output_prefix: str) -> None:
```

Comment:
We need this function to essentially enable partitions of single JSONL files. It is basically mimicking `reshard_jsonl` which was deprecated in the Dask to Ray migration: https://github.com/NVIDIA-NeMo/Curator/blob/dask/nemo_curator/scripts/make_data_shards.py

I think it is okay for just the tutorial for now, but we should add it back to the codebase eventually.

Reply:
WDYT of moving this atleast to where the other MR is being merged for `split_parquet_by_size`

Reply:
If there isn't a ticket can you create one?

Reply:
Yes, I think we can use https://github.com/NVIDIA-NeMo/Curator/issues/1156 to track this. I'll link this comment in that issue too.